airflow/serialization/serialized_objects.py

```diff
@@ -1401,6 +1401,8 @@ def serialize_task_group(cls, task_group: TaskGroup) -> dict[str, Any] | None:
                 "value": cls.serialize(expand_input.value),
             }
             encoded["is_mapped"] = True
+            if task_group.max_active_groups_per_dagrun is not None:
+                encoded["max_active_groups_per_dagrun"] = task_group.max_active_groups_per_dagrun
```

Comment:
Should this not be outside the block instead? Why does this only need to be set for mapped task groups?

Reply:
If the task group is not mapped, it will always only have one instance per dag run.
In this condition, the concurrency limitation is not needed.

Reply:
The serialisation data is important for recording purpose though (the result is shown in UI, for example), and silently dropping the value can be confusing.